### PR TITLE
Fix reposync update notice formatting and date parsing (bsc#1194447)

### PR DESCRIPTION
**Pull request by vzhestkov**
## What does this PR change?

Fixes traceback on notice format fixing for some of the repos:
```
Traceback (most recent call last):
File "/usr/lib/python3.6/site-packages/spacewalk/satellite_tools/reposync.py", line 1514, in fix_notice
notice['update_id'] = notice['update_id'] + '-' + notice['version']
TypeError: must be str, not float

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
File "/usr/lib/python3.6/site-packages/spacewalk/satellite_tools/reposync.py", line 598, in sync
self.import_updates(plugin)
File "/usr/lib/python3.6/site-packages/spacewalk/satellite_tools/reposync.py", line 736, in import_updates
processed_updates_count = self.upload_updates(notices)
File "/usr/lib/python3.6/site-packages/spacewalk/satellite_tools/reposync.py", line 935, in upload_updates
notice = self.fix_notice(notice)
File "/usr/lib/python3.6/site-packages/spacewalk/satellite_tools/reposync.py", line 1516, in fix_notice
notice._md['update_id'] = notice['update_id'] + '-' + notice['version']
TypeError: must be str, not float
```

Additionally fixes date formatting if the timestamp is specified with milliseconds.

## Documentation
- No documentation needed: only internal and user invisible changes

## Links

Fixes https://github.com/SUSE/spacewalk/issues/16648

## Changelogs

- Fix reposync update notice formatting and date parsing (bsc#1194447)

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: backend/satellite_tools/reposync.py [hunk 1]
```diff
@@ -1501,7 +1501,10 @@ def _to_db_date(date):
         if not date:
             ret = datetime.utcnow()
         elif date.isdigit():
-            ret = datetime.fromtimestamp(float(date))
+            try:
+                ret = datetime.fromtimestamp(float(date))
+            except ValueError:
+                ret = datetime.fromtimestamp(float(date)/1000)
```
[the remark this answers is not in the document]
Comment (vzhestkov) [reply backend/satellite_tools/reposync.py:1507]: In the repo I used for test the date is specified not in seconds but in milliseconds.
As the result the traceback like this one:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: year 54076 is out of range
```

File: backend/satellite_tools/reposync.py [hunk 2]
```diff
@@ -1524,9 +1527,9 @@ def fix_notice(notice):
             # old suse style; we need to append the version to id
             # to get a seperate patch for every issue
             try:
-                notice['update_id'] = notice['update_id'] + '-' + notice['version']
+                notice['update_id'] = "%s-%s" % (notice['update_id'], notice['version'])
             except TypeError: # yum in RHEL5 does not have __setitem__
-                notice._md['update_id'] = notice['update_id'] + '-' + notice['version']
+                notice._md['update_id'] = "%s-%s" % (notice['update_id'], notice['version'])
```
Comment (agraul) [inline backend/satellite_tools/reposync.py:1532]: Looking at the comment, I'd say it is safe to remove the try/except statements.

**Review by agraul — CHANGES_REQUESTED** [review]

File: backend/satellite_tools/reposync.py (hunk 2 above)
Comment (vzhestkov) [reply backend/satellite_tools/reposync.py:1532]: Do you mean because of dropping RHEL5?

Comment (vzhestkov) [reply backend/satellite_tools/reposync.py:1532]: Maybe not just dropping, but in case of Uyuni as it was never build for RHEL5 :-D

Comment (agraul) [reply backend/satellite_tools/reposync.py:1532]: Exactly, we don't need to keep backwards compatibility for RHEL5's yum

Comment (vzhestkov) [reply backend/satellite_tools/reposync.py]: I just dropped one more workaround for the same issue with https://github.com/uyuni-project/uyuni/pull/4728/commits/11d3a18f30ab77b1d1db659c7ec16edefe86e5aa

File: backend/satellite_tools/reposync.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vzhestkov) [reply backend/satellite_tools/reposync.py]: Comment is in https://github.com/uyuni-project/uyuni/pull/4728/commits/ab63ed3f53b6fd0d6e1525d493ea3c3cb6f72f79

**Pull request merged**